central/debug/service/diagnostics.go

```diff
@@ -237,14 +236,24 @@ func getClusterCandidate(conn connection.SensorConnection, clusterNameMap map[st
 	delete(clusterNameMap, clusterID)
 
 	// if there are no cluster filters, all clusters must be considered.
-	if opts.clusters != nil && sliceutils.Find(opts.clusters, clusterName) == -1 {
+	if checkClusterExcluded(opts, clusterName) {
 		return "", "", false
 	}
 
 	// Make sure we use a name that doesn't clash with any other cluster name.
 	return clusterName, sanitizeClusterName(clusterName), true
 }
 
+func checkClusterExcluded(opts debugDumpOptions, clusterName string) bool {
```

Comment:
may be a nit: feel like naming it `isClusterExcluded` would sound better, but we've been somewhat inconsistent in naming those functions.